consensus_pbft.go

```diff
@@ -412,35 +420,38 @@ func (p *Pbft) runValidateState(ctx context.Context) { // start new round
 			panic(fmt.Errorf("BUG: Unexpected message type: %s in %s", msg.Type, p.getState()))
 		}
 
-		//if voting power is disabled - count number of votes
-		if !p.IsVotingPowerEnabled() {
-			if p.state.numPrepared() > p.state.NumValid() {
-				// we have received enough prepare messages
-				sendCommit(span)
-			}
+		switch votingMetadata := p.votingMetadata.(type) {
+		case *NonWeightedVotingMetadata:
```

Comment:
I am not a fan of this switch, I would like to merge both things on the same function. This is increasing the logic by a lot.

Reply:
Any ideas how to make this more better? I'm using such `switch-case` construction on two places: `runValidateState` and `runRoundChangeState`. I could make some function, which would have two callbacks as parameters: `weightedCallbackFn` and `nonWeightedCallbackFn`. Any better idea?

Reply:
Any ideas how to make this more better? I'm using such `switch-case` construction on two places: `runValidateState` and `runRoundChangeState`. I could make some function, which would have two callbacks as parameters: `weightedCallbackFn` and `nonWeightedCallbackFn`. Any better idea?

Reply:
To me, both are the same problem. You have a given quorum weight for the sequence, and each node has a weight. In the voting power case, the weight is the staked value and the node weight is their stake. In the case of the one-peer-one-vote, the weight is the number of peers and the node weight is one. I think you could simplify this with two backend functions, one to get the quorum weight and another to query the node weight.

Reply:
Ok I see your point, makes sense, will try to abstract that part as well.